docs/components/control/clerk-loading.mdx

```diff
@@ -13,27 +13,53 @@ The `<ClerkLoading>` renders its children while Clerk is loading, and is helpful
 
 <Tabs type="framework" items={["Next.js", "React", "Remix"]}>
   <Tab>
-  ```tsx filename="pages/_app.tsx"
-  import "@/styles/globals.css";
-  import { ClerkLoaded, ClerkLoading, ClerkProvider } from "@clerk/nextjs";
-  import { AppProps } from "next/app";
-
-  function MyApp({ Component, pageProps }: AppProps) {
-    return (
-
-    <ClerkProvider {...pageProps}>
-      <ClerkLoading>
-        <div>Clerk is loading</div>
-      </ClerkLoading>
-      <ClerkLoaded>
-        <Component {...pageProps} />
-      </ClerkLoaded>
-    </ClerkProvider>
-    ); 
-  }
+    <CodeBlockTabs type="router" options={["App Router", "Pages Router"]}>
+    ```tsx filename="layout.tsx"
+      import { ClerkProvider } from '@clerk/nextjs'
+      import './globals.css';
+
+      export default function RootLayout({
+        children,
+      }: {
+        children: React.ReactNode
+      }) {
+        return (
+          <ClerkProvider>
```

Comment:
This doesn't work because the `<html>` and `<body>` tags must never be children of ClerkProvider, or any of the conditional rendering components like ClerkLoading. [More info](https://stackoverflow.com/a/76596868)

You can do this:
```
    <html lang="en">
      <body>
        <ClerkProvider>
          <ClerkLoading>
            <div>Clerk is loading...</div>
          </ClerkLoading>
          <ClerkLoaded>
            <h1>Clerk loaded</h1>
          </ClerkLoaded>
        </ClerkProvider>
      </body>
    </html>
```

Reply:
wait what, that's how we've always created our nextjs apps. its even written in the quickstarts https://clerk.com/docs/quickstarts/nextjs#add-clerk-provider-to-your-app
and is how all of the sample/quickstart apps are set up. and they work perfectly fine.

maybe for the conditional rendering components - but they definitely can be children of clerkprovider